### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by noctera**
| Status  | Type  | Env Vars Change |
| :---: | :---: | :---: |
| Ready| Enhancement| Yes |

## Description

Added logging for errors when in dev mode

## Motivation and Context

Errors will be logged now

## How Has This Been Tested?

locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (non-breaking change which cleans up / improves existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added unit tests to code changes that make sense.
- [x] I have considered the accessibility of my changes (i.e. did I add proper content descriptions to images, or run my changes with talkback enabled?)
- [x] I have documented my code if needed

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -13,3 +13,5 @@ DB_DATABASE=
 
 JWT_SECRET=
 SALT_ROUNDS=
+
+DEBUG= #true or false
```
Comment (luwol03) [inline .env.example]: why did you rename this to `DEBUG`?

Comment (noctera) [reply .env.example]: because I thought NODE_ENV shouldn't be set over .env

File: app/Middleware/ErrorMiddleware.js [hunk 2]
```diff
@@ -38,8 +39,13 @@ const errorHandler = (err, _req, res, _next) => {
   const response = {
     code: statusCode,
     message,
+    ...(process.env.DEBUG === 'true' && { stack: err.stack }),
   };
 
+  if (process.env.DEBUG === 'true') {
+    logger.error(err);
+  }
+
```
Comment (luwol03) [inline app/Middleware/ErrorMiddleware.js:48]: can we log 5xx errors in production?

File: package-lock.json [hunk 3]
```diff
@@ -11048,4 +11048,4 @@
       "dev": true
     }
   }
-}
+}
```
Comment (luwol03) [inline package-lock.json]: is that intended?

File: app/config/logger.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+const winston = require('winston');
+require('dotenv').config();
```
Comment (luwol03) [inline app/config/logger.js]: we only need this once.

File: .env.example (hunk 1 above)
Comment (luwol03) [inline .env.example]: shouldn't we move this comment in a documentation when we have one?

**Review by luwol03 — COMMENTED** [review]
Can you also migrate the consol.logs in the other files?

File: package-lock.json (hunk 3 above)
Comment (noctera) [reply package-lock.json]: no. I was just copying the othe package-lock because of a merge conflict. Seems a bit buggy here

File: app/config/logger.js (hunk 4 above)
Comment (noctera) [reply app/config/logger.js]: oh, this is still from the old code

File: .env.example (hunk 1 above)
Comment (noctera) [reply .env.example]: yes

**Review by luwol03 — APPROVED** [review]
